js/packages/cli/src/commands/upload.ts

```diff
@@ -370,22 +370,25 @@ export async function uploadV2({
     saveCache(cacheName, env, cacheContent);
   }
 
-  let uploadSuccessful = true;
+  let uploadSuccessful = false;
   if (!hiddenSettings) {
-    uploadSuccessful = await writeIndices({
-      anchorProgram,
-      cacheContent,
-      cacheName,
-      env,
-      candyMachine,
-      walletKeyPair,
-      rateLimit,
-    });
+    while (!uploadSuccessful) {
+      console.log(`Starting upload at ${new Date()}`);
+      uploadSuccessful = await writeIndices({
+        anchorProgram,
+        cacheContent,
+        cacheName,
+        env,
+        candyMachine,
+        walletKeyPair,
+        rateLimit,
+      });
 
-    const uploadedItems = Object.values(cacheContent.items).filter(
-      (f: { link: string }) => !!f.link,
-    ).length;
-    uploadSuccessful = uploadSuccessful && uploadedItems === totalNFTs;
+      const uploadedItems = Object.values(cacheContent.items).filter(
+        (f: { link: string }) => !!f.link,
+      ).length;
+      uploadSuccessful = uploadSuccessful && uploadedItems === totalNFTs;
```

Comment:
Note that your loop might not work correctly if the image upload did not succeed fully, in this case it will keep looping infinitely as image/other assets are uploaded at the top.